src/flb_network.c

```diff
@@ -186,6 +189,35 @@ int flb_net_socket_tcp_fastopen(flb_sockfd_t fd)
     return setsockopt(fd, SOL_TCP, TCP_FASTOPEN, &qlen, sizeof(qlen));
 }
 
+
+/*
+ * Enable TCP keepalive
+ */
+int flb_net_socket_tcp_keepalive(flb_sockfd_t fd, struct flb_net_setup *net)
+{
+    int ret = 0;
+    int enabled = 1;
+    int time = net->tcp_keepalive_time;
+    int intvl = net->tcp_keepalive_interval;
+    int probes = net->tcp_keepalive_probes;
+
+    if (time == 0 || intvl == 0 || probes == 0) {
+        return 0;
+    }
+
+    ret |= setsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, (void *)&enabled, sizeof(enabled));
```

Comment:
do not change every one unless there is a non-default value

Reply:
Done, no longer assumes that user would want to set all 3 values at a time